core/sr-primitives/src/generic/unchecked_mortal_compact_extrinsic.rs

```diff
@@ -101,23 +117,28 @@ where
 				CheckedExtrinsic {
 					signed: Some((signed, (raw_payload.0).0)),
 					function: raw_payload.1,
+					tip: raw_payload.2,
 				}
 			}
 			None => CheckedExtrinsic {
 				signed: None,
 				function: self.function,
+				// an unsigned transaction cannot have tips.
+				tip: Tip::None,
```

Comment:
I wonder if we should actually reject unsigned transactions that specify any `tip` other than `Tip::None` instead of silently accept them and convert the `Tip`.

Reply:
seems fair to do. I tried to be more _compatible_. I'll keep the conversation open.

Reply:
- This is okay but actually, the beneficial thing to do is to reject the transaction at the very beginning if it is unsigned and has a tip. See new changes. I think now it makes the most sense.